skimage/filters/_unsharp_mask.py

```diff
@@ -0,0 +1,104 @@
+import numpy as np
+from scipy.ndimage.filters import gaussian_filter
+from skimage.util.dtype import dtype_limits
+
+
+def _unsharp_mask_single_channel(image, radius, amount):
+    """Single channel implementation of the unsharp masking filter."""
+    blurred = gaussian_filter(image.astype(np.float_),
+                              sigma=radius,
+                              mode='reflect')
+    result = image + (image - blurred) * amount
+
+    # Enforce image dtype limits
+    if image.dtype.kind == 'f':
+        _min, _max = dtype_limits(
+            image, clip_negative=np.all(image >= 0))
+    else:
+        _min, _max = dtype_limits(image, clip_negative=False)
+    result = np.clip(result, _min, _max).astype(image.dtype)
+    return result
+
+
+def unsharp_mask(image, radius=1.0, amount=1.0, multichannel=False):
+    """Unsharp masking filter.
+
+    The sharp details are identified as the difference between the original
+    image and its blurred version. These details are then scaled, and added
+    back to the original image.
+
+    Parameters
+    ----------
+    image : (M[, N[, ..., P]][, C]) ndarray
```

Comment:
Just ndarray would be fine, since the function is n-d (cool!).

Reply:
Ping on this comment.

Reply:
I thought we agreed on this notation?

Reply:
Anything is fine by me, just let me know what do you prefer in the end.